homeassistant/components/goodwe/switch.py

```diff
@@ -0,0 +1,135 @@
+"""GoodWe PV inverter switch settings entities."""
+from __future__ import annotations
+
+from dataclasses import dataclass
+import logging
+from typing import Any
+from datetime import timedelta
+
+from goodwe import Inverter, InverterError
+
+from homeassistant.components.switch import (
+    SwitchDeviceClass,
+    SwitchEntity,
+    SwitchEntityDescription,
+)
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import EntityCategory
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.device_registry import DeviceInfo
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+
+from .const import DOMAIN, KEY_DEVICE_INFO, KEY_INVERTER
+
+_LOGGER = logging.getLogger(__name__)
+
+SCAN_INTERVAL = timedelta(minutes=2)
+
+@dataclass(frozen=True)
+class GoodweSwitchEntityDescriptionBase:
+    """Required values when describing Goodwe switch settings."""
+
+    icon_on: str
+    icon_off: str
+
+
+@dataclass(frozen=True)
```

Comment:
https://developers.home-assistant.io/blog/2023/12/11/entity-description-changes

So `kw_only=True` shoul be there too?

Reply:
added, thanks